torch/csrc/jit/passes/fuse_linear.cpp

```diff
@@ -7,11 +7,11 @@ namespace jit {
 
 void FuseLinear(std::shared_ptr<Graph>& graph) {
   std::string addmm_pattern = R"IR(
-    graph(%input, %weight_t, %bias, %beta, %alpha):
```

Comment:
Patch for this (cc @eellison):

```patch
diff --git a/test/jit/test_graph_rewrite_passes.py b/test/jit/test_graph_rewrite_passes.py
index 95bb564da7..f9cc1fc84c 100644
--- a/test/jit/test_graph_rewrite_passes.py
+++ b/test/jit/test_graph_rewrite_passes.py
@@ -59,3 +59,22 @@ class TestGraphRewritePasses(JitTestCase):
         FileCheck().check_not("aten::linear").run(model.graph)
         # make sure it runs
         model(x)
+
+    def test_fuse_addmm_to_linear(self):
+        def f(x, y, z):
+            t = z.t()
+            return torch.addmm(x, y, t)
+
+        scripted = torch.jit.script(f)
+        torch._C._jit_pass_fuse_linear(scripted.graph)
+        check_not = ["aten::matmul", "aten::addmm", "aten::add_", "aten::t("]
+        for cn in check_not:
+            FileCheck().check_not(cn).run(scripted.graph)
+
+        def g(x, y, z):
+            t = z.t()
+            return torch.addmm(x, y, t, dtype=torch.half)
+
+        scripted = torch.jit.script(g)
+        torch._C._jit_pass_fuse_linear(scripted.graph)
+        FileCheck().check("aten::addmm").run(scripted.graph)
diff --git a/torch/csrc/jit/passes/fuse_linear.cpp b/torch/csrc/jit/passes/fuse_linear.cpp
index 1fb15c860f..93b544b487 100644
--- a/torch/csrc/jit/passes/fuse_linear.cpp
+++ b/torch/csrc/jit/passes/fuse_linear.cpp
@@ -21,6 +21,14 @@ void FuseLinear(std::shared_ptr<Graph>& graph) {
     return is_int_constant(match, vmap, "beta", 1);
   };
 
+  auto dtype_is_none = [](const Match& match,
+                          const std::unordered_map<std::string, Value*>& vmap) {
+    const auto& match_vmap = match.values_map;
+    auto v = toIValue(match_vmap.at(vmap.at("dtype")));
+    auto is_none = v && v->isNone();
+    return is_none;
+  };
+
   // check %weight_t is produced by `aten::t` to make sure
   // we can transform the pattern to `aten::linear`
   auto weight_transposed =
@@ -38,7 +46,8 @@ void FuseLinear(std::shared_ptr<Graph>& graph) {
   addmm_to_linear.RegisterRewritePattern(
       addmm_pattern, fused_linear_addmm, value_mappings);
   addmm_to_linear.runOnGraph(
-      graph, {aten_add_alpha_is_one, beta_is_one, weight_transposed});
+      graph,
+      {aten_add_alpha_is_one, beta_is_one, dtype_is_none, weight_transposed});
 
   std::string matmul_add_pattern = R"IR(
     graph(%input, %weight_t, %bias, %alpha):
```

Reply:
Thanks for the patch. I applied the patch on my local machine. I'm still seeing failures for 
```
python test/test_ops.py -v -k test_variant_consistency_jit_addmm
```
Not sure if those are related.

Reply:
See my comment on symbolic_script.cpp, I think that should fix it

Reply:
Thanks! That fixed it.